python/ray/train/checkpoint.py

```diff
@@ -135,7 +140,7 @@ def on_init(self):
     def on_start_training(
         self,
         checkpoint_strategy: Optional[CheckpointStrategy],
-        run_dir: Path,
+        run_dir: Optional[Path] = None,
```

Comment:
Good point, it ultimately gets used as the `parent_path` in `construct_path`. I updated that so it can also accept None, in which case the parent path will just be cwd.

Reply:
Wait, is that actually the correct behavior though? Shouldn't no `run_dir` mean there is no checkpoint dir (i.e. it's handled externally by Tune)?

Reply:
Updated to not allow `latest_checkpoint_dir` to even be called by the DataParallelCheckpointManager!